workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/ReaderBase.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System.Collections.Generic;
+using Improbable.Worker;
+using Unity.Entities;
+
+namespace Improbable.Gdk.Core.MonoBehaviours
+{
+    public abstract class ReaderBase<TSpatialComponentData, TComponentUpdate>
+        : IReader<TSpatialComponentData>,
+            IReaderInternal
+        where TSpatialComponentData : ISpatialComponentData
+        where TComponentUpdate : ISpatialComponentUpdate<TSpatialComponentData>
+    {
+        protected readonly Unity.Entities.Entity Entity;
+        protected readonly EntityManager EntityManager;
+
+        private readonly List<GameObjectDelegates.AuthorityChanged> authorityChangedDelegates
+            = new List<GameObjectDelegates.AuthorityChanged>();
+
+        private readonly List<GameObjectDelegates.ComponentUpdated<TSpatialComponentData>> componentUpdateDelegates
+            = new List<GameObjectDelegates.ComponentUpdated<TSpatialComponentData>>();
+
+        protected ReaderBase(Unity.Entities.Entity entity, EntityManager entityManager)
+        {
+            this.Entity = entity;
+            EntityManager = entityManager;
+        }
+
+        public Authority Authority
+        {
+            get
+            {
+                if (EntityManager.HasComponent<Authoritative<TSpatialComponentData>>(Entity))
+                {
+                    return Authority.Authoritative;
+                }
+
+                if (EntityManager.HasComponent<AuthorityLossImminent<TSpatialComponentData>>(Entity))
+                {
+                    return Authority.AuthorityLossImminent;
+                }
+
+                // TODO reviewers: should this throw an error instead?
```

Comment:
We should throw a "this should never happen" sort of error for the case where no authority tag is on an entity.